primitives/core/src/crypto.rs

```diff
@@ -37,10 +37,23 @@ use regex::Regex;
 use base58::{FromBase58, ToBase58};
 #[cfg(feature = "std")]
 use crate::hexdisplay::HexDisplay;
-use zeroize::Zeroize;
 #[doc(hidden)]
 pub use sp_std::ops::Deref;
 use sp_runtime_interface::pass_by::PassByInner;
+/// Trait to zeroize a memory buffer.
+pub use zeroize::Zeroize;
+/// Trait for accessing reference to `SecretString`.
+pub use secrecy::ExposeSecret;
+/// A store for sensitive data.
+#[cfg(feature = "std")]
+pub use secrecy::SecretString;
+
+#[cfg(feature = "std")]
+/// Parse a sercret string, returning a displayable error.
+pub fn secrety_string_from_str(s: &str) -> Result<SecretString, String> {
+	Ok(std::str::FromStr::from_str(s)
+		.map_err(|_e| "Could not get SecretString".to_string())?)
+}
```

Comment:
```suggestion
/// Parse a sercret string, returning a displayable error.
#[cfg(feature = "std")]
pub fn secret_string_from_str(s: &str) -> Result<SecretString, String> {
	std::str::FromStr::from_str(s).map_err(|_e| "Could not get SecretString".to_string())
}
```

And please move this to `cli`. As we only need this in the cli. Please put it into `client/cli/src/params/mod.rs`.